src/PrestaShopBundle/Controller/Admin/AttributeController.php

```diff
@@ -174,8 +174,20 @@ public function ensureProductHasDefaultCombination(Product $product, array $comb
         if (count($combinations)) {
             $defaultProductAttributeId = $product->getDefaultIdProductAttribute();
             if (!$defaultProductAttributeId) {
-                $firstAttributeCombination = reset($combinations[0]);
-                $product->setDefaultAttribute($firstAttributeCombination['id_product_attribute']);
+                /*
+                 * Combinations indexed by position, then attribute id
+                 * ex: $combinations = [
+                 *  3 => [ //4th position attribute
+                 *      45 => [ //product_attribute id
+                 *      ]
+                 *  ]
+                 * ]
+                 */
+                $firstPosition = array_keys($combinations)[0];
+                if (isset($combinations[$firstPosition]) && count($combinations[$firstPosition])) {
```

Comment:
`!empty($combinations[$firstPosition])` will do the isset and count at the same time but faster :D

Reply:
Thanks, I always forget about this one!

Reply:
You're welcome :wink: